[BUG] errors compiling the hello world exmaple

**Describe the bug**
I get a couple of compiler errors compiling `hello.cpp` which is generated from the `hello.cpp2` example

**To Reproduce**
Steps to reproduce the behavior:
1. Running `cppfront` on `hello.cpp2` from the [documentation](https://hsutter.github.io/cppfront/welcome/hello-world/) results in
<details>
<summary>hello.cpp</summary>
#define CPP2_IMPORT_STD          Yes

//=== Cpp2 type declarations ====================================================
#include "cpp2util.h"

#line 1 "hello.cpp2"

//=== Cpp2 type definitions and function declarations ===========================

#line 1 "hello.cpp2"
auto main() -> int;

#line 7 "hello.cpp2"
[[nodiscard]] auto hello(cpp2::impl::in<std::string_view> msg) -> decltype(auto);

//=== Cpp2 function definitions =================================================

#line 1 "hello.cpp2"
auto main() -> int{
#line 2 "hello.cpp2"
    std::vector words {"Alice", "Bob"};
    hello(CPP2_ASSERT_IN_BOUNDS_LITERAL(words, 0));
    hello(CPP2_ASSERT_IN_BOUNDS_LITERAL(cpp2::move(words), 1));
}

#line 7 "hello.cpp2"
[[nodiscard]] auto hello(cpp2::impl::in<std::string_view> msg) -> decltype(auto) {
    return std::cout << "Hello, " + cpp2::to_string(msg) + "!\n";  }
</details>

2. Command lines including which C++ compiler you are using: 
g++ version 14.2.0 on Ubuntu 24.4
`g++ -std=c++20 hello.cpp -o hello -I$HOME/.local/cppfront/include`

3. Expected result - compile successfully

4. Actual result/error:

```log
hello.cpp2: In function ‘int main()’:
hello.cpp2:3:10: error: use of ‘decltype(auto) hello(cpp2::impl::in<std::basic_string_view<char> >)’ before deduction of ‘auto’
    3 |     hello( words[0] );
      |     ~~~~~^~~~~~~~~~~~~
hello.cpp2:4:10: error: use of ‘decltype(auto) hello(cpp2::impl::in<std::basic_string_view<char> >)’ before deduction of ‘auto’
    4 |     hello( words[1] );
      |     ~~~~~^~~~~~~~~~~~~
```
**Additional context**
Add any other context about the problem here.

The Clang error is good, MSVC is also good:
```
<source>(5): error C3779: 'hello': a function that returns 'decltype(auto)' cannot be used before it is defined
<source>(3): note: see declaration of 'hello'
```

The GCC error could use some clarification.  Once you know what the error means, you can squint and see the meaning there, but it's not immediately obvious.

```
use of ‘decltype(auto) hello(cpp2::impl::in<std::basic_string_view<char> >)’ before deduction of ‘auto’
```

Without the forward declaration, the errors are: 

```
GCC: error: 'hello' was not declared in this scope
Clang: error: use of undeclared identifier 'hello'
MSVC: error C3861: 'hello': identifier not found
```

So it's a tossup whether it's better or not.  I'm guessing most C++ developers are familiar with order dependence, and know what these errors mean.  

Maybe cppfront can detect the order-dependent calls and produce an error?  I suppose that would be difficult, since there could be another version that was `#include`d from somewhere that it doesn't know about.

--- a/docs/welcome/hello-world.md
+++ b/docs/welcome/hello-world.md
@@ -152,4 +152,4 @@ Hello, world!
 ### &#10148; Next: [Adding cppfront to your existing C++ project](integration.md)
 
 
-[^clean-cpp1]: For presentation purposes, this documentation generally shows the `.cpp` as generated when using cppfront's `-c` (short for `-clean-cpp1`), which suppresses extra information cppfront normally emits in the `.cpp` to light up C++ tools (e.g., to let IDEs integrate cppfront error message output, debuggers step to the right lines in Cpp2 source code, and so forth). In normal use, you won't need or even want `-c`.
+[^clean-cpp1]: For presentation purposes, this documentation generally shows the `.cpp` as generated when using cppfront's `-cl` (short for `-clean-cpp1`), which suppresses extra information cppfront normally emits in the `.cpp` to light up C++ tools (e.g., to let IDEs integrate cppfront error message output, debuggers step to the right lines in Cpp2 source code, and so forth). In normal use, you won't need or even want `-cl`.

--- a/docs/welcome/hello-world.md
+++ b/docs/welcome/hello-world.md
@@ -24,8 +24,9 @@ main: () = {
     hello( words[1] );
 }
 
-hello: (msg: std::string_view) =
+hello: (msg: std::string_view) = {
     std::cout << "Hello, (msg)$!\n";
+}
 ```
 
 This short program code already illustrates a few Cpp2 essentials.

